lib/api/controllers/adminController.js

```diff
@@ -46,53 +41,31 @@ class AdminController extends BaseController {
       'loadSecurities',
       'resetCache',
       'resetDatabase',
-      'resetKuzzleData',
       'resetSecurity',
       'shutdown'
     ]);
   }
-  
-
-  /**
-   * Reset the internal storage components (internalEngine index, cache and memory storage)
-   */
-  resetKuzzleData (request) {
-    this._lockAction(request, 'Kuzzle is already resetting his data.');
-
-    return this.kuzzle.repositories.user.truncate({ refresh: 'wait_for' })
-      .then(() => this.kuzzle.internalEngine.deleteIndex())
-      .then(() => this.kuzzle.services.list.internalCache.flushdb())
-      .then(() => {
-        this.kuzzle.indexCache.remove(this.kuzzle.internalEngine.index);
-
-        return this.kuzzle.internalEngine.bootstrap.all();
-      })
-      .then(() => {
-        delete _locks.resetKuzzleData;
-
-        return { acknowledge: true };
-      })
-      .catch(error => {
-        delete _locks.resetKuzzleData;
-        throw error;
-      });
-  }
 
   /**
    * Reset Redis cache
    */
   resetCache (request) {
-    assertArgsHasAttribute(request, 'database');
+    const database = this.getStringParam(request, 'args.database');
 
-    const
-      database = request.input.args.database,
-      cacheEngine = this.kuzzle.services.list[database];
+    let cacheEngine;
 
-    if (!cacheEngine || typeof cacheEngine.flushdb !== 'function') {
+    // @todo allow only memoryStorage
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)